src/Ushahidi/Modules/V5/Policies/RolePolicy.php

```diff
@@ -0,0 +1,107 @@
+<?php
+
+namespace Ushahidi\Modules\V5\Policies;
+
+use Ushahidi\Modules\V5\Models\Role;
+use App\Auth\GenericUser as User;
+use Ushahidi\Core\Concerns\AdminAccess;
+use Ushahidi\Core\Concerns\PrivAccess;
+use Ushahidi\Core\Concerns\UserContext;
+
+class RolePolicy
+{
+
+
+    use UserContext;
+
+    // It uses `PrivAccess` to provide the `getAllowedPrivs` method.
+    use PrivAccess;
+
+    // Check if user has Admin access
+    use AdminAccess;
+
+    protected $user;
+
+    /**
+     * @param User $user
+     * @return bool
+     */
+    public function index(User $user):bool
+    {
+        $empty_role = new Role();
+        return $this->isAllowed($empty_role, 'search', $user);
+    }
+
+    /**
+     * @param User $user
+     * @param Role $role
+     * @return bool
+     */
+    public function show(User $user, Role $role):bool
+    {
+        return $this->isAllowed($role, 'read', $user);
+    }
+
+    /**
+     * @param User $user
+     * @param Role $role
+     * @return bool
+     */
+    public function delete(User $user, Role $role):bool
+    {
+        return $this->isAllowed($role, 'delete', $user);
+    }
+    /**
+     * @param User $user
+     * @param Role $role
+     * @return bool
+     */
+    public function update(User $user, Role $role):bool
+    {
+        return $this->isAllowed($role, 'update', $user);
+    }
+
+
+    /**
+     * @param User $user
+     * @param Role $role
+     * @return bool
+     */
+    public function store(User $user):bool
+    {
+        $role = new Role();
+        return $this->isAllowed($role, 'create', $user);
+    }
+
+    /**
+     * @param Role $role
+     * @param string $privilege
+     * @param user $user
+     * @return bool
+     */
+    public function isAllowed($role, $privilege, $user = null):bool
+    {
+        
+        $authorizer = service('authorizer.role');
+        $user = $authorizer->getUser();
```

Comment:
You're overwriting $user passed as an argument to this method.

Reply:
$authorizer->getUser(); return entity user while the passed user is Generic user 
I know it's confusing  .. this will be solved after we merge  Wisdom PRs so we can use the entity directly.